### PR TITLE
fix(forge): add coverage to test setUp

### PR DESCRIPTION
**Pull request by Inphi**
## Motivation

Improve forge coverage reporting of functions called during test `setUp`. This allows initializer functions, but not ctors, called during setUp (ex: https://github.com/foundry-rs/foundry/issues/4553) to be covered.

## Solution

The `TestSetup` is extended to include coverage info. The `setUp` coverage is then merged with `HitMaps` generated by test runners.

Consider the coverage reporting of the `CounterTest:test_Increment` test generated by `forge init`. Prior to this change, we see that `Counter.setNumber` isn't covered because it's only called during `setUp`:
image

With this patch, we get full test coverage in `Counter.sol`:
image

### CODE REVIEW

**Review by Evalir — COMMENTED** [review]
This looks reasonable—Is it possible for you to provide a screenshot of this working when using lcov or something similar for vis? Just to ensure everything looks okay.

**Review by Evalir — CHANGES_REQUESTED** [review]
this will also need rebasing—sorry about this, lots of moving parts rn, we'll be done soon!6

**Comment by Inphi** [comment]
@Evalir Thanks for the review. PR's been rebased and I've added a visual description of this change.

File: crates/forge/src/runner.rs [hunk 1]
```diff
@@ -638,10 +652,21 @@ impl<'a> ContractRunner<'a> {
...
-            coverage: result.coverage,
+            coverage,
             labeled_addresses,
             debug,
             breakpoints,
         }
     }
 }
+
+// Utility function to merge coverage options
```
Comment (Evalir) [inline crates/forge/src/runner.rs:663]:
```suggestion
/// Utility function to merge coverage options
```

**Review by Evalir — CHANGES_REQUESTED** [review]
nice, looks reasonable, and tests look good. Pending @mattsse and nit

**Review by Evalir — APPROVED** [review]

**Pull request merged**